include/RAJA/policy/cuda/raja_cudaerrchk.hpp

```diff
@@ -25,16 +25,82 @@
 #if defined(RAJA_ENABLE_CUDA)
 
 #include <iostream>
+#include <utility>
+#include <algorithm>
+#include <tuple>
+#include <array>
+#include <string_view>
 #include <string>
+#include <sstream>
+#include <stdexcept>
 
 #include <cuda.h>
 #include <cuda_runtime.h>
 
+#include "RAJA/util/Printing.hpp"
 #include "RAJA/util/macros.hpp"
+#include "RAJA/util/for_each.hpp"
+
+#include "cub/util_type.cuh"
 
 namespace RAJA
 {
 
+namespace detail
+{
+
+template < >
+struct StreamInsertHelper<CUDA_DIM_T&>
+{
+  CUDA_DIM_T& m_val;
+
+  std::ostream& operator()(std::ostream& str) const
+  {
+    return str << "{" << m_val.x
+               << "," << m_val.y
+               << "," << m_val.z
+               << "}";
+  }
+};
+///
+template < >
+struct StreamInsertHelper<CUDA_DIM_T const&>
+{
+  CUDA_DIM_T const& m_val;
+
+  std::ostream& operator()(std::ostream& str) const
+  {
+    return str << "{" << m_val.x
+               << "," << m_val.y
+               << "," << m_val.z
+               << "}";
+  }
+};
```

Comment:
Its not ideal that I have to provide a const and non-const ref version of each printer. I did this because `cub::DoubleBuffer` has non-const member functions.

Reply:
I could always take by const& and then const_cast the const away for `cub::DoubleBuffer`. Do people think that is reasonable?